### PR TITLE
Automate unit tests in Git

### PR DESCRIPTION
**Pull request by Matheusstreisky**

### CODE REVIEW

File: .github/workflows/dev.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Dev branch CI
+
+on:
+  pull_request:
+    branches:
+      - dev
```
Comment (JoaoPucci) [inline .github/workflows/dev.yml:7]: As a next step, might be worth setting up a CI for the `main` branch too.

File: .github/workflows/dev.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+      - name: Cache Maven dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-maven-
+
+      - name: Clean and build
+        run: mvn -B -e clean install -DskipTests
```
Comment (JoaoPucci) [inline .github/workflows/dev.yml:31]: By running `clean` every time, isn't the cache being deleted?

Comment (Matheusstreisky) [reply .github/workflows/dev.yml:31]: No. I run the CI again and the dependences not be downloaded. Was used using the cache.

File: .github/workflows/dev.yml (hunk 1 above)
Comment (Matheusstreisky) [reply .github/workflows/dev.yml:7]: Updated the file to inclues the main branch.

**Pull request merged**